network/manager.go

```diff
@@ -710,21 +712,21 @@ func cnsEndpointInfotoCNIEpInfo(endpointInfo restserver.EndpointInfo, endpointID
 		epInfo.IfName = ifName
 		epInfo.HostIfName = ipInfo.HostVethName
 		epInfo.HNSEndpointID = ipInfo.HnsEndpointID
+		epInfo.HNSNetworkID = ipInfo.HnsNetworkID
+		epInfo.MacAddress = net.HardwareAddr(ipInfo.MacAddress)
 	}
 	return epInfo
 }
 
+// generateCNSIPInfoMap generates a CNS ifNametoIPInfoMap structure based on CNI endpoint
 func generateCNSIPInfoMap(ep *endpoint) map[string]*restserver.IPInfo {
 	ifNametoIPInfoMap := make(map[string]*restserver.IPInfo) // key : interface name, value : IPInfo
-	if ep.IfName != "" {
-		ifNametoIPInfoMap[ep.IfName].NICType = cns.InfraNIC
-		ifNametoIPInfoMap[ep.IfName].HnsEndpointID = ep.HnsId
-		ifNametoIPInfoMap[ep.IfName].HostVethName = ep.HostIfName
-	}
-	if ep.SecondaryInterfaces != nil {
-		for ifName, InterfaceInfo := range ep.SecondaryInterfaces {
-			ifNametoIPInfoMap[ifName].NICType = InterfaceInfo.NICType
-		}
+	ifNametoIPInfoMap[ep.IfName] = &restserver.IPInfo{
+		NICType:       cns.InfraNIC,
```

Comment:
does the code break if we do `ep.NICType` ?

Reply:
I do not see any NICType in ep. If it's created and filled out then it can be used